board/app/controllers/thread_controller.php

```diff
@@ -1,107 +1,141 @@
-            break;
-        }  
+                break;
+            }  
 
         $this->set(get_defined_vars());
-        $this->render($current_page);
+        $this->render($current_page);               
     }
- 
-    //Displays maximum number of threads per page
+          
+    //Displays the comments of the thread
+    public function view() 
+    {
+        $per_page = self::MAX_COMMENT_PER_PAGE;
+        $current_page = Param::get('page', 1);
+        $pagination = new SimplePagination($current_page, $per_page);      
+
+        $thread_id = Param::get('thread_id'); 
+        $thread = Thread::get($thread_id); 
+        $comments = Comment::getAll($pagination->start_index -1, $pagination->count + 1, $thread_id);
+                        
+        $total = Comment::countAll($thread_id);
+        $pagination->checkLastPage($comments);
+        $pages = ceil($total / $per_page);
+        $this->set(get_defined_vars());
+    }
+
+    /*
+    * Sorting of threads
+    */
+
+    //All Threads
     public function index() 
     {
         $per_page = self::MAX_THREAD_PER_PAGE; 
         $current_page = Param::get('page', 1);
-        $pagination = new SimplePagination($current_page, $per_page) ;
-
+        $pagination = new SimplePagination($current_page, $per_page);
         $threads = Thread::getAll($pagination->start_index -1, $pagination->count + 1);
         $pagination->checkLastPage($threads);
-
         $total = Thread::CountAll();
         $pages = ceil($total / $per_page);
-    
         $this->set(get_defined_vars()); 
-    }   
+    }  
 
-    //Displays the comments of the thread
-    public function view() 
+    public function mythreads()
     {
-        $per_page = self::MAX_COMMENT_PER_PAGE;
+        $per_page = self::MAX_THREAD_PER_PAGE;
         $current_page = Param::get('page', 1);
-        $pagination = new SimplePagination($current_page, $per_page) ;
+        $pagination = new SimplePagination($current_page, $per_page);
 
-        $thread = Thread::get(Param::get('thread_id'));
-        $comments = $thread->getComments($pagination->start_index -1, $pagination->count + 1);
-        
-        $pagination->checkLastPage($comments);
-        $total = $thread->countComments();
+        $id = $_SESSION['user_id'];
+        $threads = Thread::getAllMyThread($pagination->start_index -1, $pagination->count + 1, $id);
+        $pagination->checkLastPage($threads);
+
+        $total = Thread::countAllThreadByUserId($_SESSION['user_id']);
         $pages = ceil($total / $per_page);
+
         $this->set(get_defined_vars());
+        $this->render('my_threads');
     }
 
-    //Write a new comment
-    public function write()
+    public function bycategory() 
     {
-        $thread = Thread::get(Param::get('thread_id'));
-        $comment = new Comment();
-        $current_page = Param::get('page_next', 'write');
+        $category = Param::get('category','none');
+            if ( $category !== 'none') {
```

Comment:
:scream: 
Hehe, done po. Thank you :bow:
-->https://github.com/maragatchalian/FoodFacts/commit/96d26e82c9eff60c909847a48eb896ba79346448